devito/core/operator.py

```diff
@@ -208,3 +212,68 @@ def _specialize_iet(cls, graph, **kwargs):
             passes_mapper['linearize'](graph)
 
         return graph
+
+
+# Wrappers for optimization options
+
+
+class OptOption(object):
+    pass
+
+
+class ParTileArg(tuple):
+
+    def __new__(cls, items, shm=0, tag=None):
+        obj = super().__new__(cls, items)
+        obj.shm = shm
+        obj.tag = tag
+        return obj
+
+
+class ParTile(tuple, OptOption):
+
+    def __new__(cls, items, default=None):
+        if not items:
```

Comment:
Can all these conditionals be simplified?

Reply:
not really, it enables multiple input modes